### PR TITLE
Add tooltips to all Icon buttons #633

### LINKED ISSUE #633: Add tooltips to all Icon buttons
For buttons that are an icon only with no text.

### PR DESCRIPTION
**Pull request by asuresh-code**
## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking

connect to #633 
closes #633

### CODE REVIEW

**Comment by asuresh-code** [comment]
I've just noticed I missed some icons; sorting it out now.

File: src/catalogue/category/catalogueCard.component.tsx
Comment (joelvdavies) [inline src/catalogue/category/catalogueCard.component.tsx:89]: Just a question, why was this needed? I can't see much difference between them.

Comment (asuresh-code) [reply src/catalogue/category/catalogueCard.component.tsx:89]: All of that code was to access the offset and change the position of the tooltips. It wasn't strictly needed, but I tried to move the tooltips to not block other text.

I realise my laptop's display is smaller, so the offset was noticeable on my end, but not others.

Comment (joelvdavies) [reply src/catalogue/category/catalogueCard.component.tsx:89]: Maybe its just me but I cant make them overlap without them

image

Do you have an example you could screenshot?

Otherwise I would probably just remove the offsets - the main thing is that it doesn't cover the component its attached to otherwise the user can just move the mouse again. It saves us having to maintain individual offsets across the site.

Comment (asuresh-code) [reply src/catalogue/category/catalogueCard.component.tsx:89]: w/o the offset
image

w/ the offset
image

However, It was never blocking its own component, so in that case it should be fine to remove the offsets.

Comment (joelvdavies) [reply src/catalogue/category/catalogueCard.component.tsx:89]: Yeah I would just remove it and leave it to the tooltip component to decide where to place it then. It was good thought though.

**Review by joelvdavies — CHANGES_REQUESTED** [review]
LGTM - I just noticed there are there are some warnings in the tests if you look at the CI (https://github.com/ral-facilities/inventory-management-system/actions/runs/10403795349/job/28810966203). These should be fixed before merging.

**Comment by asuresh-code** [comment]
> LGTM - I just noticed there are there are some warnings in the tests if you look at the CI (https://github.com/ral-facilities/inventory-management-system/actions/runs/10403795349/job/28810966203). These should be fixed before merging.

I've address the issues regarding missing spans for the Tooltip. However, I haven't been able to fix the warning about not wrapping with act().

Adding act() to [waitFor()](https://github.com/ral-facilities/inventory-management-system/actions/runs/10453014822/job/28942635952) or [waitFor() and modifyValues()](https://github.com/ral-facilities/inventory-management-system/actions/runs/10451913627/job/28939268344) causes the lint to fail, and adding it just to [modifyValues](https://github.com/ral-facilities/inventory-management-system/actions/runs/10452798376/job/28942287471) causes a timeout in waitFor()

**Comment by asuresh-code** [comment]
It seems that adding a tooltip to the Home icon in Breadcrumbs is creating the warning in 2 tests:

catalogueCategoryDialog.component.test.tsx
- Add Catalogue Category Dialog
      - create a catalogue category with content being catalogue items
      - create a catalogue category with content being catalogue items (allowed_values list of numbers)

It's not obvious why this is the case, since a similar test is not creating the warning: create a catalogue category with content being catalogue items (allowed_values list of strings)

**Comment by asuresh-code** [comment]
image
Update: This warning seems to be caused by some tooltips, not just the Home Breadcrumbs one.

**Review by joshuadkitenge — CHANGES_REQUESTED** [review]
Looks good. It would be great if the breadcrumb home tooltip title could be made configurable, allowing it to be changed to 'Catalogue Home,' 'Systems Home,' 'Admin Home,' or 'Manufacturers Home.'

File: src/view/breadcrumbs.component.tsx
Comment (joshuadkitenge) [inline src/view/breadcrumbs.component.tsx:48]: It would be better to change the navigateHomeAriaLabel prop in the breadcrumb component to homeLocation: 'System' | 'Admin' | .... We can then use homeLocation to construct both the aria-label and the tooltip title. This approach will ensure that the aria-labels are more consistent.

**Review by joshuadkitenge — CHANGES_REQUESTED** [review]

File: src/admin/admin.component.tsx
Comment (joelvdavies) [inline src/admin/admin.component.tsx:72]:
```suggestion
              homeLocation="Admin"
```

File: src/catalogue/catalogue.component.tsx
Comment (joelvdavies) [inline src/catalogue/catalogue.component.tsx:314]:
```suggestion
              homeLocation="Catalogue"
```

File: src/catalogue/category/catalogueCategoryDirectoryDialog.component.tsx
Comment (joelvdavies) [inline src/catalogue/category/catalogueCategoryDirectoryDialog.component.tsx:179]:
```suggestion
              homeLocation="Catalogue"
```

File: src/catalogue/items/catalogueItemDirectoryDialog.component.tsx
Comment (joelvdavies) [inline src/catalogue/items/catalogueItemDirectoryDialog.component.tsx:192]:
```suggestion
              homeLocation="Catalogue"
```
There could be a few others, just a picky styling comment.

**Review by joelvdavies — APPROVED** [review]
LGTM - Probably worth squashing and merging this one.

**Review by joshuadkitenge — APPROVED** [review]

**Pull request merged**
